### PR TITLE
KeyExchangeContext interfaces

### PR DESCRIPTION
**Pull request by public**

### CODE REVIEW

**Comment by public** [comment]
Should this move up into `asymmetric/interfaces.py`?

**Comment by public** [comment]
Moved to `kex.py`

File: src/cryptography/hazmat/primitives/asymmetric/dh.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 from __future__ import absolute_import, division, print_function
 
+import abc
+
```
Comment (alex) [inline src/cryptography/hazmat/primitives/asymmetric/dh.py]: Unused.

Comment (public) [inline src/cryptography/hazmat/primitives/asymmetric/dh.py]: Removed.

**Comment by public** [comment]
retest this please

File: src/cryptography/hazmat/primitives/asymmetric/kex.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# This file is dual licensed under the terms of the Apache License, Version
```
Comment (alex) [inline src/cryptography/hazmat/primitives/asymmetric/kex.py]: What do you think about naming this file `key_exchange.py`?

Comment (public) [inline src/cryptography/hazmat/primitives/asymmetric/kex.py]: I would be OK with that.

**Comment by public** [comment]
retest this please

File: src/cryptography/hazmat/primitives/asymmetric/kex.py (hunk 2 above)
Comment (reaperhulk) [inline src/cryptography/hazmat/primitives/asymmetric/kex.py]: Looks like this is still an outstanding item on this PR.

**Comment by reaperhulk** [comment]
this needs a rebase

**Comment by public** [comment]
Rebased.

**Comment by public** [comment]
Need to rename that kex file still though :)

**Comment by reaperhulk** [comment]
closing in favor of #1882

**Pull request closed without merging**